docs/index.md

```diff
@@ -1179,6 +1179,47 @@ app/components
 </div>
 ```
 
+##### Translations
+
+In order to support `I18n` translations scoped to the component you can include `ViewComponent::SidecarI18n`
+into your component:
+
+`app/components/example_component.rb`
+
+```ruby
+module ExampleComponent < ApplicationComponent
+  include ViewComponent::SidecarI18n
+end
+```
+
+This allows for adding a "sidecar" YAML file that will define translations for the component:
+
+`app/components/example_component.yml`
+
+```yml
+en:
+  # This key will be expanded to `en.example_component.hello`
```

Comment:
What happens if there are multiple `example_component`s? Perhaps we need to hash this identifier?